pcdsdevices/epics_motor.py

```diff
@@ -735,6 +738,31 @@ def flag_is_cleared(value=None, **kwargs):
             status_wait(st, timeout=timeout)
         return st
 
+    @property
+    def md(self):
+        if self._md is None:
+            raise AttributeError('Device does not have an attached md, '
+                                 'and was likely not initialized from happi')
+        return self._md
+
+    @md.setter
+    def md(self, new_md):
+        """ initialize attributes when md is set """
+        self._md = new_md
+        self._extra = self._md.extraneous
+        self._id = self._extra.get('_id')
+        self._pvbase = self._extra['pvbase']
```

Comment:
There's one more dict access here, this one could also raise a `KeyError`.
I suggest either changing this to a .get or using self.prefix instead.
```suggestion
        self._pvbase = self._extra.get('pvbase')
```